src/pool/cli.c

```diff
@@ -217,6 +217,57 @@ dc_pool_choose_svc_rank(const uuid_t puuid, struct rsvc_client *cli,
 	return rc;
 }
 
+/* Choose a pool service replica rank by label. If the rsvc module indicates
+ * DER_NOTREPLICA, (clients only) try to refresh the list by querying the MS.
+ */
+int
+dc_pool_choose_svc_rank_bylabel(const char *label, uuid_t puuid,
```

Comment:
I can consider this next, though I may be tempted to reduce duplication similarly in dc_mgmt_get_pool_svc_ranks() / dc_mgmt_pool_find_bylabel() and ultimately across the CaRT RPCs GET_SVCRANKS and FIND_BYLABEL. And maybe into the dRPC upcalls on the engine/server side. I think all of this would (if some unforeseen complexities are not seen along the way) result in a single "find pool by label or UUID" approach. At which point we might decide to abandon the existing GET_SVCRANKS interface that is by UUID only (or, perhaps leave it on the engine side in case older clients interact with newer servers?).

Reply:
Yeah, I thought about the similar dup between dc_mgmt_get_pool_svc_ranks and dc_mgmt_pool_find_bylabel. If they use different request and reply types, combining them might be awkward. Don't try _too_ hard, I think.